tutor/src/models/course.js

```diff
@@ -106,10 +107,42 @@ export default class Course extends BaseModel {
     return !!find(this.roles, 'isTeacher');
   }
 
+  @computed get taskPlans() {
+    if (this.isTeacher) {
+      return TeacherTaskPlans.forCourseId(this.id).values();
+    }
+
+    return [];
+  }
+
+  @computed get hasReadingPublishing() {
+    if (this.isTeacher) {
+      return some(this.taskPlans, {is_publishing: true, type: 'reading'});
+    }
+
+    return false;
+  }
+
+  @computed get hasHomeworkPublishing() {
+    if (this.isTeacher) {
+      return some(this.taskPlans, {is_publishing: true, type: 'homework'});
+    }
+
+    return false;
+  }
+
```

Comment:
I like the addition of `taskPlans`, but since `TeacherTaskPlans` extends our custom `Map` class, we could write filter chaining functions for it like courses has (https://github.com/openstax/tutor-js/blob/master/tutor/src/models/courses-map.js#L29-L35)  that way we could do things like: `course.taskPlans.reading.isEmpty`   I think we're ok to not check if the user is a teacher since if they're not they can't request task plans anyway and they'll always be empty.

Also why the two blank lines before `return` ?  I guess it does make the return stand out a bit..